src/daft-connect/src/functions/core.rs

```diff
@@ -62,6 +64,26 @@ impl FunctionModule for CoreFunctions {
     }
 }
 
+pub struct WhenFunction;
+impl SparkFunction for WhenFunction {
+    fn to_expr(&self, args: &[Expression]) -> ConnectResult<daft_dsl::ExprRef> {
+        let args = args
+            .iter()
+            .map(analyze_expr)
+            .collect::<ConnectResult<Vec<_>>>()?;
+
+        let predicate = args[0].clone();
+        let if_false = args[1].clone();
+        let if_true = args.get(2).cloned().unwrap_or_else(null_lit);
```

Comment:
there should only be two arguments right? and `if_false` would just always be null.

https://spark.apache.org/docs/latest/api/python/reference/pyspark.sql/api/pyspark.sql.functions.when.html

Reply:
wild but sounds good